examples/ggez/main.rs

```diff
@@ -0,0 +1,130 @@
+use map::MapHandler;
+use tiled::{Map, FilesystemResourceCache};
+
+fn main() -> GameResult {
+    // init ggez
+    let cb = ggez::ContextBuilder::new("rs-tiled + ggez", "rs-tiled")
+        .window_setup(ggez::conf::WindowSetup::default()
+            .title("rs-tiled + ggez example")
+            .vsync(false)
+        )
+        .window_mode(ggez::conf::WindowMode::default()
+            .dimensions(1000.0, 800.0)
+        )
+        .add_resource_path(""); // add repo root to ggez filesystem (our example map looks for `assets/tilesheet.png`)
+
+    let (mut ctx, event_loop) = cb.build()?;
+
+    // construct and start the Game
+    let state = Game::new(&mut ctx)?;
+    event::run(ctx, event_loop, state)
+}
+
+struct Game {
+    map: MapHandler,
+    pan: (f32, f32),
+    scale: f32,
+}
+
+impl Game {
+    fn new(ctx: &mut ggez::Context) -> GameResult<Self> {
+        graphics::set_default_filter(ctx, graphics::FilterMode::Nearest);
+
+        // load the map
+        let mut cache = FilesystemResourceCache::new();
+        let map = Map::parse_file("assets/tiled_base64_external.tmx", &mut cache).unwrap();
+
+        let map_handler = MapHandler::new(map, ctx).unwrap();
+
+        Ok(Self {
+            map: map_handler,
+            pan: (0.0, 0.0),
+            scale: 1.0,
+        })
+    }
+}
+
+impl event::EventHandler<ggez::GameError> for Game {
+    fn update(&mut self, _ctx: &mut Context) -> GameResult {
+        Ok(())
+    }
+
+    fn draw(&mut self, ctx: &mut Context) -> GameResult {
+        // fill background color
+        let bg_color: ggez::graphics::Color = self.map.background_color().unwrap_or([0.1, 0.2, 0.3, 1.0].into());
+        graphics::clear(ctx, bg_color);
+
+        // draw map tiles + objects
+
+        let window_size = graphics::size(ctx);
+
+        let draw_param = DrawParam::default()
+            .dest([
+                self.pan.0 + window_size.0 / 2.0 - (self.map.width() * self.map.tile_width()) as f32 / 2.0, 
+                self.pan.1 + window_size.1 / 2.0 - (self.map.height() * self.map.tile_height()) as f32 / 2.0
+                ])
+            .scale([self.scale, self.scale]);
+
+        self.map.draw(ctx, draw_param, self.pan)?;
+
+        // draw map bounds
+
+        let rect = self.map.get_bounds();
+        let r1 =
+            graphics::Mesh::new_rectangle(ctx, graphics::DrawMode::stroke(2.0 / self.scale), rect, graphics::Color::from_rgb_u32(0x888888))?;
+        graphics::draw(ctx, &r1, draw_param)?;
+
+        // draw fps
+
+        let fps = ggez::timer::fps(ctx);
+        let text = graphics::Text::new(format!("{:.0} fps", fps));
+
+        graphics::draw(
+            ctx,
+            &text,
+            DrawParam::default()
+                .dest([window_size.0 - text.width(ctx) - 40.0, 10.0])
+                .scale([1.25, 1.25])
+                .color(graphics::Color::WHITE),
+        )?;
```

Comment:
You can replace these comments followed by code by functions; The end result will be cleaner and easier to follow